eva-accession-clustering-automation/create_clustering_properties.py

```diff
@@ -0,0 +1,186 @@
+
+
+def add_spring_properties(properties):
+    properties.write('\n')
+    properties.write(
+        '#See https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.1-Release-Notes#bean-overriding' + '\n')
+    properties.write('spring.main.allow-bean-definition-overriding=true' + '\n')
+    properties.write(
+        '#As this is a spring batch application, disable the embedded tomcat. This is the new way to do that for spring 2.' + '\n')
+    properties.write('spring.main.web-application-type=none' + '\n')
+    properties.write('\n')
+    properties.write(
+        '# This entry is put just to avoid a warning message in the logs when you start the spring-boot application.' + '\n')
+    properties.write(
+        '# This bug is from hibernate which tries to retrieve some metadata from postgresql db and failed to find that and logs as a warning' + '\n')
+    properties.write('# It doesnt cause any issue though.' + '\n')
+    properties.write('spring.jpa.properties.hibernate.jdbc.lob.non_contextual_creation = true' + '\n')
+
+
+def add_mongo_properties(property_line, properties):
+    property_line.write('\n')
+    mongo_hosts_and_ports = str(properties['eva.mongo.host'])
+    mongo_uri = 'mongodb://' + mongo_hosts_and_ports
+    property_line.write('spring.data.mongodb.uri=' + mongo_uri + '\n')
+    property_line.write('spring.data.mongodb.database=' + str(properties['eva.accession.mongo.database']) + '\n')
+    property_line.write('spring.data.mongodb.username=' + str(properties['eva.mongo.user']) + '\n')
+    property_line.write('spring.data.mongodb.password=' + str(properties['eva.mongo.passwd']) + '\n')
+    property_line.write('mongodb.read-preference=primary' + '\n')
+
+
+def add_job_tracker_properties(property_line, properties):
+    property_line.write('\n')
+    property_line.write('spring.datasource.driver-class-name=org.postgresql.Driver' + '\n')
+    property_line.write('spring.datasource.url=' + str(properties['eva.accession.jdbc.url']) + '\n')
+    property_line.write('spring.datasource.username=' + str(properties['eva.accession.user']) + '\n')
+    property_line.write('spring.datasource.password=' + str(properties['eva.accession.password']) + '\n')
+    property_line.write('spring.datasource.tomcat.max-active=3' + '\n')
+
+
+def check_valid_sources(source):
+    """
+    This method checks that ony MONGO and VCF are used as sources
+    """
+    if source.upper() not in ('MONGO', 'VCF'):
+        logger.error("Wrong source specified. Please choose between MONGO and VCF")
+        sys.exit(1)
+
+
+def check_vcf_source_requirements(source, vcf_file, project_accession):
+    """
+    This method checks that if the source is VCF the VCF file and project accession are provided
+    """
+    if source.upper() == 'VCF' and not (vcf_file and project_accession):
+        logger.error('If the source is VCF the file path and project accession must be provided')
+        sys.exit(1)
```

Comment:
Not sure if you have thought hard about this, but in general I would throw an exception in case some caller knows how to recover from this or wants to do something extra before exiting the program.